lib/node_modules/@stdlib/array/base/cusome/lib/assign.js

```diff
@@ -0,0 +1,312 @@
+}
+
+/**
+* Cumulatively tests whether at least n array elements in accessor array are truthy.
+*
+* @private
+* @param {Object} x - input array object
+* @param {integer} n - number of elements to be truthy
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = toAccessorArray( [ false, false, false, true, true ] );
+* var n=2;
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, true, null ]);
+*
+* var arr = accessors( arraylike2object( x ), n, arraylike2object( y ), 2, 0 );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns false
+*
+* v = y.get( 6 );
+* // returns false
+*
+* v = y.get( 8 );
+* // returns true
+*/
+function accessors( x, n, y, stride, offset ) {
+	var xdata;
+	var ydata;
+	var xget;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	xdata = x.data;
+	ydata = y.data;
+
+	xget = x.accessors[ 0 ];
+	yset = y.accessors[ 1 ];
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < xdata.length; i++ ) {
+		if ( flg === false && xget( xdata, i ) && n > 0 ) {
+			n -= 1;
+		}
+		if (n === 0) {
+			flg = true;
+		}
+		yset( ydata, io, flg );
+		io += stride;
+	}
+	return ydata;
+}
+
+/**
+* Cumulatively tests whether at least n array elements in a provided complex number are truthy and assigns results to provided output array.
+*
+* @private
+* @param {Collection} x - array containing interleaved real and imaginary components
+* @param {integer} n - number of elements to be truthy
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Float64Array( [ 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0 ] );
+* var n = 2;
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+* var arr = complex( x, n, arraylike2object( y ), 2, 0 );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns false
+*
+* v = y.get( 6 );
+* // returns false
+*
+* v = y.get( 8 );
+* // returns false
+*/
+function complex( x, n, y, stride, offset ) {
+	var ydata;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	yset = y.accessors[ 1 ];
+	ydata = y.data;
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < x.length; i += 2 ) {
+		if ( flg === false && n > 0 && ( x[ i ] || x[ i+1 ] ) ) {
```

Comment:
Aren't the `flg === false` and `n > 0` checks redundant?

Reply:
@kgryte 
when n=0  the result should be all trues . what about writing in following way
```
	for ( i = 0; i < x.length; i++ ) {
		if ( flg === false && x[ i ] ) {
			n -= 1;
			if (n <= 0) {
				flg = true;
			}
		}
		yset( ydata, io, flg );
		io += stride;
	}
```

Reply:
Yes, that should work.